File: api/src/main/java/br/ifsul/lp3/messagesender/api/service/user/UserService.java
package br.ifsul.lp3.messagesender.api.service.user;

import br.ifsul.lp3.messagesender.api.web.controller.authentication.request.RegisterAuthenticationRequest;
import br.ifsul.lp3.messagesender.api.web.controller.user.response.UserResponse;
import org.springframework.data.domain.Page;

public interface UserService {

    void save(RegisterAuthenticationRequest request);

    Page<UserResponse> findAll(int page);

    Page<UserResponse> findAllByUsernameAutoComplete(String username, int page);

    UserResponse findLoggedUser();
}

File: web-app/src/services/user.service.ts
import { api } from "./api.service";
import { UserWithID, Paged } from "../model";

export const userService = {
    getPagedUsersByUsername: async ({ page, username }: { page: number, username: string }) => {
        const response = await api.get<Paged<UserWithID>>(`/user/find-by-username/autocomplete/${username}?page=${page}`)
        return response.data
    },
    getUsersByPage: async (page: number) => {
        const response = await api.get<Paged<UserWithID>>(`/user?page=${page}`)
        return response.data
    },
    getAuthUser: async () => {
        const response = await api.get<UserWithID>('/user/find-logged-user')
        return response.data
    }
}